CMakeLists.txt

```diff
@@ -28,6 +31,8 @@ foreach(lib ${rabit_libs})
   target_include_directories(${lib} PUBLIC
     "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>"
     "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include/rabit>"
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/dmlc-core/include>"
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/dmlc-core/include/dmlc>"
```

Comment:
One my machine introducing this line causes all sorts of errors. Without it it builds fine. Did you try building it on your end?
```suggestion
```

Reply:
```
cq-C02WC09QHTDG:rabit cq$ ls
CMakeLists.txt	Makefile	cmake		doc		include		python		src
LICENSE		README.md	dmlc-core	guide		lib		scripts		test
cq-C02WC09QHTDG:rabit cq$ mkdir build
cq-C02WC09QHTDG:rabit cq$ cd build/
cq-C02WC09QHTDG:build cq$ ls
cq-C02WC09QHTDG:build cq$ cmake ..
-- The C compiler identification is AppleClang 10.0.0.10001044
-- The CXX compiler identification is AppleClang 10.0.0.10001044
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/cq/xgboost/rabit/build
cq-C02WC09QHTDG:build cq$ make
Scanning dependencies of target rabit
[  8%] Building CXX object CMakeFiles/rabit.dir/src/allreduce_base.cc.o
[ 16%] Building CXX object CMakeFiles/rabit.dir/src/allreduce_robust.cc.o
[ 25%] Building CXX object CMakeFiles/rabit.dir/src/engine.cc.o
[ 33%] Building CXX object CMakeFiles/rabit.dir/src/c_api.cc.o
[ 41%] Linking CXX static library librabit.a
[ 41%] Built target rabit
Scanning dependencies of target rabit_empty
[ 50%] Building CXX object CMakeFiles/rabit_empty.dir/src/engine_empty.cc.o
[ 58%] Building CXX object CMakeFiles/rabit_empty.dir/src/c_api.cc.o
[ 66%] Linking CXX static library librabit_empty.a
[ 66%] Built target rabit_empty
Scanning dependencies of target rabit_base
[ 75%] Building CXX object CMakeFiles/rabit_base.dir/src/allreduce_base.cc.o
[ 83%] Building CXX object CMakeFiles/rabit_base.dir/src/engine_base.cc.o
[ 91%] Building CXX object CMakeFiles/rabit_base.dir/src/c_api.cc.o
[100%] Linking CXX static library librabit_base.a
[100%] Built target rabit_base
```

Reply:
addressed here 
https://github.com/dmlc/rabit/pull/86/commits/37497ff5386b83498c8699f7fd0e5da7e10f4277

Reply:
Any idea what the cause was? Was it double importing the headers? Maybe gcc specific?